app/jobs/cache_analytics_job.rb

```diff
@@ -5,38 +5,25 @@ class CacheAnalyticsJob < ApplicationJob
   queue_as :default
 
   def perform(*_args)
-    Analytic.transaction do
-      analytics_by_jur_path = {}
-      root_jurs = []
+    Jurisdiction.find_each do |jur|
+      Analytic.transaction do
+        analytic = Analytic.new(jurisdiction_id: jur.id)
 
-      leaf_nodes = Jurisdiction.all.select { |jur| jur.has_children? == false }
-      leaf_nodes.each do |leaf_jurisdiction|
-        leaf_analytic = self.class.calculate_analytic_local_to_jurisdiction(leaf_jurisdiction)
-        analytics_by_jur_path[leaf_jurisdiction[:path]] = leaf_analytic
-        # Start recursive bubble up of analytic data
-        self.class.add_analytic_to_parent(leaf_jurisdiction, leaf_analytic, analytics_by_jur_path)
-      end
-
-      # Map data will be on the top-level jurisdiction only
-      root_nodes = Jurisdiction.where(ancestry: nil)
-      root_nodes.each do |root_jurisdiction|
-        symp_by_state = root_jurisdiction.all_patients.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
-        monitored_by_state = root_jurisdiction.all_patients.symptomatic.uniq.pluck(:address_state).each_with_object(Hash.new(0)) do |state, counts|
-          counts[state] += 1
+        # Map data will be on the top-level jurisdiction only (will be removed once new maps are implemented)
+        if jur.root?
+          symp_by_state = jur.all_patients.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
+          monitored_by_state = jur.all_patients.symptomatic.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
+          analytic.symptomatic_state_map = symp_by_state.to_s
+          analytic.monitoree_state_map = monitored_by_state.to_s
```

Comment:
Do you have these queries and assignments backwards?

Monitored is being assigned symptomatic query. If not, variable names are confusing!

Reply:
good catch, I just took a look at what's currently in master and it seems like it's switched there as well:

https://github.com/SaraAlert/SaraAlert/blob/master/app/jobs/cache_analytics_job.rb#L23

but when it's used in the analytics controller, it's switched again, making it correct in the end:

https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/analytics_controller.rb#L59

which I agree is super confusing.

However, if we do change this right now, it'll cause problems because past analytics are already switched and since we're taking this out soon, I think we should just leave it the way it is

Reply:
Understandable